compiler/rustc_mir/src/const_eval/eval_queries.rs

```diff
@@ -292,6 +294,18 @@ pub fn eval_to_allocation_raw_provider<'tcx>(
         if let Some(error_reported) = tcx.mir_const_qualif_opt_const_arg(def).error_occured {
             return Err(ErrorHandled::Reported(error_reported));
         }
+
+        if cid.promoted.is_none() {
```

Comment:
🤣 I did not suggest this on purpose. It's such a hack, but it should be fine (or at least not worse than the status quo)

Reply:
not sure if we are just missing the appropriate tests to still fail with an incremental ICE on a non-promoted. So this is probably not the best way to proceed

Reply:
yea I figure it's not actually a correct fix lmao